### PR TITLE
Imp/gatsby source GitHub raw integration

### PR DESCRIPTION
**Pull request by patricksimonian**
## Summary
This is a Devhub V3 effort. In an effort to utilize multiple plugins, i'm experimenting using the gatsby mapping feature against a Devhub Collection node and GithubRaw nodes. 

This pr will serve as an integration branch as i do not wish to merge in experimental code, even if its tested and stable.

### CODE REVIEW

File: app-web/gatsby/utils/githubRaw.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Created by Patrick Simonian
+*/
+const { isRegistryJson } = require('./validators');
+const flatten = require('lodash/flatten');
+
+const getFilesFromRegistry = getNodes => {
```
Comment (ShellyXueHan) [inline app-web/gatsby/utils/githubRaw.js]: where is getNodes coming from? a gatsby thing?

File: app-web/gatsby/utils/githubRaw.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const nodes = getNodes();
+  // get RegistryJson nodes
+  const registry = nodes.filter(isRegistryJson);
+  // spit out all github sources from the registry while maintaining references to the
+  // topics they belong too
+  const gitsources = registry.reduce((sources, registryItem) => {
+    const gitsources = registryItem.sourceProperties.sources.filter(s => s.sourceType === 'github');
```
Comment (ShellyXueHan) [inline app-web/gatsby/utils/githubRaw.js]: constant for 'github'

File: app-web/gatsby/utils/githubRaw.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    sources.push([gitsources, registryItem.name]);
```
Comment (ShellyXueHan) [inline app-web/gatsby/utils/githubRaw.js]: why need to specify the name? isn't registryItem.name in gitsources already?

File: app-web/gatsby/utils/githubRaw.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      }
+      return [];
+    });
+    // flatten out all urls since a single source can produce many urls
+    return [flatten(urls), topic];
+  });
+  // convert into the files array required by the gatsby github raw plugin
+  // [[url1, url2], topic] => [{url: url1, topic}, {url: url2, topic}]
+  return flatten(
+    resolvedGitSources.map(([urls, topic]) => urls.map(url => ({ url, topics: [topic] }))),
```
Comment (ShellyXueHan) [inline app-web/gatsby/utils/githubRaw.js]: when would a url have multiple topics?

**Comment by patricksimonian** [comment]
As seen on issue #518 

> After a lot of discovery, I've found that leveraging the features of gatsby itself are more than enough to accomplish 90% of our functionality (if not more). The open PR #719, is left open as leveraging 3rd party plugins and making the connections between them has introduced some dependancies that need to be worked on.
> 
> These dependancies are:
> 
> * `gatsby-source-github-raw` anchor tag transformation is required so that any links inside of markdown files are appropriately pointing to devhub pages when they exist
> * `gatsby-plugin-elastic-lunr` search engine needs to be reconfigured to work with pure markdown files
> 
> ## So what has been done?
> Extensive experimentation and proof of concepts (all documented, and even illustrated!) on how to connect 3rd party nodes together.

**Pull request merged**